server/routes/gameRoutes.test.js

```diff
@@ -0,0 +1,90 @@
+import { it, expect, describe, beforeEach } from "vitest";
+import request from "supertest";
+import app from "../index.js";
+import db from "../db/connection.js";
+
+beforeEach(async () => {
+  await db.execute(`DELETE from games`);
+
+  await db.execute(`
+    INSERT INTO games (id, name, image_url) VALUES
+        (1, 'Game 1', '/example.jpg'),
+        (2, 'Game 2', '/example.jpg'),
+        (3, 'Game 3', '/example.jpg');
+  `);
+});
+
+describe("GET /games", () => {
+  it("Responds with a list of all games", async () => {
+    const response = await request(app).get("/games");
+
+    expect(response.status).toBe(200);
+    expect(response.body).toEqual([
+      {
+        id: 1,
+        name: "Game 1",
+        image_url: "/example.jpg",
+      },
+      {
+        id: 2,
+        name: "Game 2",
+        image_url: "/example.jpg",
+      },
+      {
+        id: 3,
+        name: "Game 3",
+        image_url: "/example.jpg",
+      },
+    ]);
+  });
+});
+
+describe("GET /games/:id", () => {
+  it("Responds with a single game", async () => {
+    const response = await request(app).get("/games/1");
+
+    expect(response.status).toBe(200);
+    expect(response.body).toEqual({
+      id: 1,
+      name: "Game 1",
+      image_url: "/example.jpg",
+    });
+  });
+});
+
+describe("POST /games", () => {
+  it("Fails given an invalid body", async () => {
+    const body = {};
+
+    const response = await request(app).post("/games").send(body);
+
+    expect(response.status).toBeGreaterThanOrEqual(400);
```

Comment:
This caught my eye, doesn't feel right that we're giving a 5XX error if the user doesn't provide a body ([here](https://github.com/michaelti/play-up/blob/testing/server/controllers/gameController.js#L27-L36))

Should be a 4XX error since the client made a mistake right?

In which case, rather than `toBeGreaterThanOrEqual(400)` it could be `toBe(401)` or something similar.

Don't think this is a huge issue, it's more of a REST guideline thing.

Reply:
You're right. This endpoint should be validating the request body and responding with a 400.

I've improved the endpoint and updated the test to reflect that.